ruby_on_rails/rspec.md

```diff
@@ -112,21 +112,26 @@ Check that you see `1+2=3` in each app.
 
 ## Optional: catch javascript errors
 
-If you want to catch Javascript errors in your system tests, you can add the following to the `rails_helper.rb`:
+If you want to catch Javascript errors in your system tests, you can create `support/javascript_error_collector.rb`:
 
 ```ruby
-config.after(:each, type: :system, js: true) do
-  errors = page.driver.browser.manage.logs.get(:browser)
-  if errors.present?
+module JavaScriptErrorCollector
+  def collect_errors
+    errors = page.driver.browser.manage.logs.get(:browser)
+
     aggregate_failures 'javascript errors' do
       errors.each do |error|
         expect(error.level).not_to eq('SEVERE'), error.message
-        next unless error.level == 'WARNING'
-
-        warn 'WARN: javascript warning'
-        warn error.message
+        expect(error.level).not_to eq('WARNING'), error.message
```

Comment:
It seems to me that with this change the tests will fail also for warnings, while before they were failing for errors and just print a message for warnings. correct?

Reply:
Yes this is true. In my opinion the warnings were lost in all the console output. And javascript warnings can be important so I therefore made the tests fail. Proposal: Don't let the tests fail but make the warnings more apparent. With colors or something.

Reply:
I think we should either fail on warnings or show nothing. Warnings tend to pollute logs without getting fixed.

I'm generally on the page that warnings were in control of should make the tests fail. **BUT:** are we really in control of the warnings? What I would not want, is that tests start failing because of a Chromedriver update.

Reply:
What if we let the tests fail if the warning comes from one of the projects scripts or the browser and ignore it if it comes from a dependancy?

Reply:
I generally like the idea.

But:

* Can you even do that? How clear is the source of a warning in JS?
* The problem with Chrome updates remains: if Chrome introduces new warning after an update, our builds will suddenly fail.

Reply:
It's possible but probably difficult to account for all possibilities/sources.

At this point I think it's best to go back to just print them. Is everyone ok with that?
@coorasse @schmijos